### PR TITLE
added integration with Vertex AI

### PR DESCRIPTION
**Pull request by lkuligin**
I added Google Vertex AI models for the following use cases:
- LLM
- chat (single-turn and multi-turn chat)
- embeddings

In order to use them, you need to install VertexAI SDK (that is is the private preview) and be whitelisted to list the models themselves:
```
gsutil cp gs://vertex_sdk_llm_private_releases/SDK/google_cloud_aiplatform-1.25.dev20230413+language.models-py2.py3-none-any.whl .
pip install invoke
pip install google_cloud_aiplatform-1.25.dev20230413+language.models-py2.py3-none-any.whl "shapely<2.0.0"
```

Your end-user credentials would be used to make calls (make sure you've run `gcloud auth login` first).

### CODE REVIEW

File: langchain/chat_models/vertexai.py [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+                raise ValueError("Message should be from human if it's the first one!")
+            context, prompt = None, message.content
+        elif len(messages) == 2:
+            first_message, second_message = messages[0], messages[1]
+            if not isinstance(first_message, SystemMessage):
+                raise ValueError(
+                    "The first message should be a system one if there're two of them."
+                )
+            if not isinstance(second_message, HumanMessage):
+                raise ValueError("The second message should from human!")
```
Comment (dev2049) [inline langchain/chat_models/vertexai.py:92]: should be from a

File: langchain/llms/vertex.py [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+    def _enforce_stop_words(self, text: str, stop: Optional[List[str]]) -> str:
+        if stop:
+            return enforce_stop_tokens(text, stop)
+        return text
+
+    @property
+    def _llm_type(self) -> str:
+        """Return type of llm."""
+        return self.client._MODEL_NAME
```
Comment (dev2049) [inline langchain/llms/vertex.py:46]: this should be something like `vertex` or `vertexai`. the `model_name` attribute can be used for getting the actual model name

**Review by dev2049 — COMMENTED** [review]
amazing! couple of comments

**Comment by MarcDAFrame** [comment]
what's the status of this @dev2049, I can address the comments if this is still a WIP?

**Comment by lkuligin** [comment]
@MarcDAFrame we're working on a new one that would incorporate the latest changes in the SDK.

**Pull request closed without merging**